### ISSUE TITLE
Regular domain admin account cannot change nicAdapter “Request Failed (431)”

### ISSUE THREAD

**Issue by kricud** [apache/cloudstack#5724]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.15.2
~~~

##### SUMMARY

Regular domain admin account cannot change nicAdapter “Request Failed (431)” in new UI in OLD it works.
![acs_nic](https://user-images.githubusercontent.com/30048481/143592942-5635862e-7976-4aef-88fb-a6e8afb80cd6.png)

##### STEPS TO REPRODUCE

~~~
Edit nicAdapter settings
~~~

##### EXPECTED RESULTS

~~~
Regular domain admin account can change nicAdapter
~~~

##### ACTUAL RESULTS

~~~
Regular domain admin account cannot change nicAdapter
~~~

**Comment by shwstppr** [comment]
@kricud are you sure about that this worked in the old UI. The error seen here comes from backend and I checked in 4.15.1.0 as well and the code is same - it doesn't allow non-root admins to update these settings,
https://github.com/apache/cloudstack/blob/4.15.1.0/server/src/main/java/com/cloud/vm/UserVmManagerImpl.java#L2574-L2583

**Comment by kricud** [comment]
@shwstppr ACS 4.15.2
![E1000ToVMxnet3_error](https://user-images.githubusercontent.com/30048481/148955298-d3fd03a3-cbe3-4afc-8292-56de440ebc0b.JPG)
From top:
First legacy UI
Second new UI

**Comment by shwstppr** [comment]
@kricud seems to be working okay in my testing for `nicAdapter` with domain admin on 4.15.2.0
![domain-admin](https://user-images.githubusercontent.com/153340/148975922-2a4b1270-9934-4f46-8ac0-4e0ae188e69f.gif)

**Comment by sureshanaparti** [comment]
@kricud @shwstppr Can you check the details set in the global settings: 'user.vm.denied.details' and 'user.vm.readonly.details'.

**Comment by kricud** [comment]
@sureshanaparti In global settings I have only user.vm.readonly.ui.details with value dataDiskController, rootDiskController. Don’t have user.vm.denied.details.

@shwstppr User admin has Role “Root Admin” Role Type “Admin” with these permissions it works.
Its not working if you have Account with Role “Domain Admin” Role Type “Domain Admin” and user created under this account cannot change nicAdapter.

**Comment by shwstppr** [comment]
@kricud in the video I was first logged in as domain admin only. Please see initial few seconds. Later I logged in as root admin to show ACS version

**Comment by kricud** [comment]
@shwstppr Step by step how to reproduce it.
1. Install Windows Server from ISO
2. Hypervisor is Vmware
3. Install VMware Tools on VM
4. Reboot
5. By default in VMware nic adapter is E1000
6. Shutdown VM
7. Try change nic adapter to VMxnet3, get and get error in new UI
8. Try change nic adapter to VMxnet3 in old UI and its working.
![1](https://user-images.githubusercontent.com/30048481/149099133-5f9455fd-5065-4c1d-9d03-e2bcf367273f.png)
![2](https://user-images.githubusercontent.com/30048481/149099143-32b34af6-c6e7-40e6-aada-930a2587edd5.png)
![3](https://user-images.githubusercontent.com/30048481/149099150-bc9c9153-f271-4d39-a83c-4c8648a4bc7d.png)
![4](https://user-images.githubusercontent.com/30048481/149099165-7ce9743c-b81b-4a19-a759-6eb4bc586065.png)
![55](https://user-images.githubusercontent.com/30048481/149099388-bc958f8f-48cc-4b17-819f-cdb40345a8be.png)

**Comment by sureshanaparti** [comment]
Hi @kricud Can you try with API/cmk update virtualmachine. Also, please share the API calls (from browser developer tools) from old and new UI.

**Comment by kricud** [comment]
@sureshanaparti From CMK it works

![11](https://user-images.githubusercontent.com/30048481/150669997-1c63a869-b140-4a62-b7d3-2afad979c78b.png)
![22](https://user-images.githubusercontent.com/30048481/150669998-cf14bf0b-65e4-4ca7-910c-6d2739e50838.png)
![33](https://user-images.githubusercontent.com/30048481/150669999-b97fb76c-9a2f-48ec-9014-349a9b9a36b7.png)

**Comment by sureshanaparti** [comment]
Hi @kricud Noticed the read-only detail '_rootDiskController_' is passed in the updateVirtualMachine API call from your tests, due to which API fails. I tried the same in my env, and couldn't reproduce it.

UI filters the read-only details before calling the API, but in your case the read-only detail '_rootDiskController_ exists in the call. Can you clear the browser cache and retry? Any custom UI changes in your env?

**Comment by kricud** [comment]
@sureshanaparti Cache reset is not helping
![11](https://user-images.githubusercontent.com/30048481/150750425-af270e6b-8656-4acb-9fec-8b9ba56926ad.png)
g

**Comment by shwstppr** [comment]
@kricud I've created a PR (https://github.com/apache/cloudstack/pull/5887) to fix the issue. It seems UI was not filtering readonly details

**Comment by sureshanaparti** [comment]
Fixed in #5887
